stackslib/src/net/api/getsortition.rs

```diff
@@ -0,0 +1,304 @@
+// Copyright (C) 2024 Stacks Open Internet Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::io::{Read, Seek, SeekFrom, Write};
+use std::{fs, io};
+
+use regex::{Captures, Regex};
+use serde::de::Error as de_Error;
+use stacks_common::codec::{StacksMessageCodec, MAX_MESSAGE_LEN};
+use stacks_common::types::chainstate::{
+    BlockHeaderHash, BurnchainHeaderHash, ConsensusHash, SortitionId, StacksBlockId,
+};
+use stacks_common::types::net::PeerHost;
+use stacks_common::util::hash::{to_hex, Hash160};
+use stacks_common::util::HexError;
+use {serde, serde_json};
+
+use crate::chainstate::burn::db::sortdb::SortitionDB;
+use crate::chainstate::nakamoto::{NakamotoBlock, NakamotoChainState, NakamotoStagingBlocksConn};
+use crate::chainstate::stacks::db::StacksChainState;
+use crate::chainstate::stacks::Error as ChainError;
+use crate::net::api::getblock_v3::NakamotoBlockStream;
+use crate::net::api::{prefix_hex, prefix_opt_hex};
+use crate::net::http::{
+    parse_bytes, parse_json, Error, HttpBadRequest, HttpChunkGenerator, HttpContentType,
+    HttpNotFound, HttpRequest, HttpRequestContents, HttpRequestPreamble, HttpResponse,
+    HttpResponseContents, HttpResponsePayload, HttpResponsePreamble, HttpServerError, HttpVersion,
+};
+use crate::net::httpcore::{
+    HttpRequestContentsExtensions, RPCRequestHandler, StacksHttp, StacksHttpRequest,
+    StacksHttpResponse,
+};
+use crate::net::{Error as NetError, StacksNodeState, TipRequest, MAX_HEADERS};
+use crate::util_lib::db::{DBConn, Error as DBError};
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum QuerySpecifier {
+    ConsensusHash(ConsensusHash),
+    BurnchainHeaderHash(BurnchainHeaderHash),
+    BlockHeight(u64),
+    Latest,
+}
+
+pub static RPC_SORTITION_INFO_PATH: &str = "/v3/sortition";
```

Comment:
Can this be `/v3/sortitions` (plural)?  It's more RESTful.

Reply:
Sure

Reply:
Done in 7b1ad1e